src/pybel_tools/analysis/spia.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+
+"""This module contains the BEL exporter to SPIA format.
+(see: https://bioconductor.org/packages/release/bioc/html/SPIA.html)
+"""
+from itertools import product
+from typing import Dict
+
+import pandas as pd
+
+from pybel import BELGraph
+from pybel.constants import ASSOCIATION, INCREASES, DECREASES, DIRECTLY_DECREASES, DIRECTLY_INCREASES
+from pybel.dsl.node_classes import BaseAbundance, CentralDogma, Gene, ListAbundance, Rna, ProteinModification
+from pybel_tools.constants import KEGG_RELATIONS
+
+
+def get_matrix_index(graph: BELGraph) -> set:
+    """Return set of HGNC names from Proteins/Rnas/Genes/miRNA, nodes that can be used by SPIA.
+
+    :param graph: BELGraph
+    :return: node names
+    """
+    # TODO: Using HGNC Symbols for now
+    return {
+        node.name
+        for node in graph.nodes()
+        if issubclass(type(node), BaseAbundance) and node.namespace == 'HGNC'
```

Comment:
This is a strange way of doing this... Why not`isinstance(node, BaseAbundance)`?

Reply:
It is the same but ok...